### PR TITLE
fix(client): json stringify overflow

### LINKED ISSUE #12221: Prisma generate fails on big schema file
### Bug description

Hello, my team has found some problems when using prisma generate in our project. When we run `npx prisma generate` the client responds with an error when stringifying a JSON object, but we can't see much in the error stack.

We tried running prisma generate in the original version of the project (prisma 2.26.0) and with the latest version (3.10.0 at the day of publishing this issue) 

Output when running with version 2.26.0 and DEBUG="*":
```
Environment variables loaded from .env
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : windows

C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify>dotenv npx prisma generate
  prisma:loadEnv project root found at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\package.json +0ms
  prisma:tryLoadEnv Environment variables loaded from C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\.env +0ms
[dotenv][DEBUG] "DEBUG" is already defined in `process.env` and will not be overwritten
Environment variables loaded from .env
  prisma:engines using NAPI: false +0ms
  prisma:engines binaries to download query-engine, migration-engine, introspection-engine, prisma-fmt +0ms
Prisma schema loaded from prisma\schema.prisma
  prisma:getConfig Using Query Engine Binary at: C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\query-engine-windows.exe +0ms
  prisma:getDMMF Using Query Engine Binary at: C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\query-engine-windows.exe +0ms
  prisma:generator baseDir C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\prisma +0ms
  prisma:generator prismaClientDir C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client +1ms
  prisma:GeneratorProcess 2022-03-09T13:09:52.904Z prisma:client:generator requiredEngine: queryEngine +0ms
  prisma:getGenerators neededVersions {        
  "9b816b3aa13cc270074f172f30d6eda8a8ce867d": {
    "engines": [
      "queryEngine"
    ],
    "binaryTargets": []
  }
} +0ms
  prisma:getGenerators {
  prisma:getGenerators   generatorBinaryPaths: {
  prisma:getGenerators     queryEngine: {
  prisma:getGenerators       windows: 'C:\\Users\\Admin\\Desktop\\Proyectos\\tests\\test-prisma-stringify\\node_modules\\prisma\\query-engine-windows.exe'
  prisma:getGenerators     }
  prisma:getGenerators   }
  prisma:getGenerators } +180ms
  prisma:GeneratorProcess 2022-03-09T13:10:02.791Z prisma:loadEnv skipping package.json at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\.prisma\package.json +10s
  prisma:GeneratorProcess 2022-03-09T13:10:02.792Z prisma:loadEnv skipping package.json at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\package.json +0ms
  prisma:GeneratorProcess 2022-03-09T13:10:02.793Z prisma:loadEnv project root found at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\package.json +1ms       
  prisma:tryLoadEnv Environment variables loaded from C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\.env +0ms
[dotenv][DEBUG] "DEBUG" is already defined in `process.env` and will not be overwritten
Error: Error:
RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at respond (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client\generator-build\index.js:45985:24)
    at LineStream.<anonymous> (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client\generator-build\index.js:45931:11)


    at _Generate.parse (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\build\index.js:65473:17)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async main (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\build\index.js:110797:18)
```

Output running with version 3.10.0 with DEBUG="*":
```
  prisma:engines binaries to download libquery-engine, migration-engine, introspection-engine, prisma-fmt +0ms
  prisma:loadEnv project root found at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\package.json +0ms
  prisma:tryLoadEnv Environment variables loaded from C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\.env +0ms
Environment variables loaded from .env
Prisma schema loaded from prisma\schema.prisma
  prisma:getConfig Using CLI Query Engine (Node-API Library) at: C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\query_engine-windows.dll.node +0ms
  prisma:getDMMF Using CLI Query Engine (Node-API) at: C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\query_engine-windows.dll.node +0ms
  prisma:generator prismaCLIDir C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma +0ms
  prisma:generator prismaClientDir C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client +0ms
  prisma:generator baseDir C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\prisma +1ms
  prisma:generator typescriptPath undefined +2ms
  prisma:GeneratorProcess 2022-03-09T13:17:40.083Z prisma:client:generator requiredEngine: queryEngine +0ms
  prisma:getGenerators neededVersions {        
  "73e60b76d394f8d37d8ebd1f8918c79029f0db86": {
    "engines": [
      "queryEngine"
    ],
    "binaryTargets": []
  }
} +0ms
  prisma:getGenerators {
  prisma:getGenerators   generatorBinaryPaths: {
  prisma:getGenerators     queryEngine: {
  prisma:getGenerators       windows: 'C:\\Users\\Admin\\Desktop\\Proyectos\\tests\\test-prisma-stringify\\node_modules\\prisma\\query-engine-windows.exe'
  prisma:getGenerators     }
  prisma:getGenerators   }
  prisma:getGenerators } +212ms
  prisma:GeneratorProcess 2022-03-09T13:17:49.699Z prisma:loadEnv project root found at C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\package.json +10s
  prisma:tryLoadEnv Environment variables loaded from C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\.env +0ms
Error: Error: 
RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at respond (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client\generator-build\index.js:54245:22)
    at LineStream.<anonymous> (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\@prisma\client\generator-build\index.js:54187:9)


    at Object.parse (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\build\index.js:107298:15)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
    at async main (C:\Users\Admin\Desktop\Proyectos\tests\test-prisma-stringify\node_modules\prisma\build\index.js:108740:18)
```

We will be glad to give you more information if you need it. Thanks for such an amazing ORM

### How to reproduce

1. Create a new npm project
2. Install prisma and prisma client (`npm i prisma` and `npm i -D @prisma/client`)
3. Run `npx prisma generate` with a big schema.prisma file (ours has more than 3400 lines and about 162k characters without spaces)

### Expected behavior

The prisma client should be able to generate the models from our schema

### Prisma information

Unfortunately we can't give you the schema.prisma that we area using right now. We think that with the information we are giving you, it should be possible to reproduce the same error.

The schema.prisma file has about 3400 lines and 162k characters with a total of 136 models in it.

### Environment & setup

We tried with different setups
- OS: Windows 64 bits and Linux (node image in docker)
- Database: PostgreSQL
- Node.js version: v14.17.1, v16.14.0 and v14.19.0

### Prisma Version

Executed npx prisma -v
The project originally uses prisma version 2, we tried with the latest version but we got the same results

Prisma 2.26.0:
```
prisma               : 2.26.0
@prisma/client       : 2.26.0
Current platform     : windows
Query Engine         : query-engine 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\query-engine-windows.exe)
Migration Engine     : migration-engine-cli 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine : introspection-core 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary        : prisma-fmt 9b816b3aa13cc270074f172f30d6eda8a8ce867d (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash : 9b816b3aa13cc270074f172f30d6eda8a8ce867d
Studio               : 0.408.0
```

Prisma 3.10.0:
```
prisma                  : 3.10.0
@prisma/client          : 3.10.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules\@prisma\engines\query_engine-windows.dll.node)
Migration Engine        : migration-engine-cli 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules\@prisma\engines\migration-engine-windows.exe)
Introspection Engine    : introspection-core 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules\@prisma\engines\introspection-engine-windows.exe)
Format Binary           : prisma-fmt 73e60b76d394f8d37d8ebd1f8918c79029f0db86 (at node_modules\@prisma\engines\prisma-fmt-windows.exe)
Default Engines Hash    : 73e60b76d394f8d37d8ebd1f8918c79029f0db86
Studio                  : 0.458.0
```

### PR DESCRIPTION
**Pull request by millsp**
Fixes* the generation issue discussed during the deep dive.

Next steps: add a test with a huge schema

(Review not really needed, but linking you here if you want to dig a bit more :))

fixes https://github.com/prisma/prisma/issues/12221

### CODE REVIEW

**Comment by janpio** [comment]
Note: This might or might not be in used by Studio, which might or might not be tested.

**Comment by millsp** [comment]
I found one usage of the `fileMap` here https://github.com/prisma/studio-code/blob/77a017c6b403b0d1ff62acda37f533c037cc6469/packages/pcw/src/pcw.ts/#L270. (We have alternative ways of getting the types via the DMMF that don't involve generators).

**Comment by millsp** [comment]
[That code does not look in use anymore](https://cs.github.com/?scope=org%3Aprisma&scopeName=prisma&q=getPrismaClientTypes) maybe it can be removed. Was this always internal, or is it exposed to users in some way, and how can I make sure if it's ok to remove? (cc @luanvdw @jasonkuhrt)

**Comment by jasonkuhrt** [comment]
@millsp I can't access that link, but TIL, and I joined the wait list :D 

Not sure about the world here, but can speak to the PDP use-case. All runtime PCW stuff that was in studio I've removed and it now lives in PDP (to be moved to reflector soon).

I would be surprised if the studio PCW code was used outside of Prisma.

**Comment by millsp** [comment]
Thanks! Then we just need to figure things out for studio, if this is dead code (https://github.com/search?q=getPrismaClientTypes&type=code)

**Comment by millsp** [comment]
Yes please, that would be very helpful. Thanks. We already have the proposal from Joël which would work to buy us some more time, let's see if we can fully remove this.

**Comment by jasonkuhrt** [comment]
CC @petradonka @prisma/devprod-eng

**Comment by Jolg42** [comment]
See investigation result in https://github.com/prisma/studio-code/issues/397#issuecomment-1117503240

[...]

> Conclusion:
> I cannot speak for other projects where this may have been used, and determining what can be affected by a change such as this is always sensitive but taking the above into account I feel positive that the code in question can be removed.

Studio PR bump: https://github.com/prisma/prisma/pull/13167
- The code has been removed from Studio [here](https://github.com/prisma/studio-code/pull/398)
- Studio version has been bumped and the release has been published

File: packages/client/src/generation/generateClient.ts [hunk 1]
```diff
@@ -146,7 +146,7 @@ export async function generateClient({
   clientVersion,
   engineVersion,
   activeProvider,
-}: GenerateClientOptions): Promise<BuildClientResult | undefined> {
+}: GenerateClientOptions): Promise<{}> {
```
Comment (aqrln) [inline packages/client/src/generation/generateClient.ts:149]: Does it make sense to use just `Promise<void>` at this point?

**Review by janpio — APPROVED** [review]
👍 (But scary still)

File: packages/client/src/generation/generateClient.ts (hunk 1 above)
Comment (Jolg42) [reply packages/client/src/generation/generateClient.ts:149]: Could make sense indeed.
Let's get this merged today asap though.

**Review by Jolg42 — APPROVED** [review]
🐘

**Review by aqrln — APPROVED** [review]

**Pull request merged**